### ISSUE TITLE
libtorrent dependency problematic?

### ISSUE THREAD

**Issue by maxnoe** [rucio/rucio#6853]
### Description

I was installing the dev dependencies in a fresh virtual env, which happened to use python 3.11.

This failed, as rucio has `libtorrent` in its server dependencies and `libtorrent` does not have a wheel for 3.11 and doesn't provide a source distribution.

This seems very concerning: no source distribution is available and the last update to the wheels was last year.

### Steps to reproduce

`pip install -r requirements/requirements.dev.txt` using python > 3.10

**Comment by rdimaio** [comment]
Found this relevant issue: https://github.com/arvidn/libtorrent/issues/7650

AFAIK no one is actively using the Rucio functionality that depends on `libtorrent`, so in the worst case scenario we could deprecate it, but it would be a shame. It looks like they're actively working on fixing this issue: https://github.com/arvidn/libtorrent/pull/7672, so we could try waiting some more time. Let's keep this issue open for reference

**Comment by bari12** [comment]
Yes, `libtorrent` is a secondary dependency which on a normal install is only pulled in if (optionally) specified. It's a transfer prototype which was built on bittorrent. To be honest, I would not want to take this out of the code, if they are actively trying to address this. For a normal rucio deployment this issue won't show up, because nobody uses this functionality. It shows in the dev environment because there we pull in all secondary dependencies.

**Comment by rdimaio** [comment]
Reopening cause this is still an issue - Rucio installations on Python > 3.10 will have issues installing dependencies until this is resolved

https://github.com/arvidn/libtorrent/issues/7683

**Comment by rdimaio** [comment]
https://github.com/arvidn/libtorrent/pull/7727 was merged, so this will likely be fixed soon

**Comment by rdimaio** [comment]
still no Python 3.12 wheels

**Comment by rdimaio** [comment]
Still no wheels - I pinged the main issue to ask if we can help in some way

**Comment by rdimaio** [comment]
2.0.11 has been released https://pypi.org/project/libtorrent/2.0.11/
